api/src/main/java/org/motechproject/nms/api/web/CallDetailsController.java

```diff
@@ -186,13 +193,31 @@ private String validateCallDetailsCommonElements(CallDetailRecordRequest callDet
     private String validateCallDetailsMobileKunjiElements(CallDetailRecordRequest callDetailRecordRequest) {
         StringBuilder failureReasons = new StringBuilder();
 
-        if (null == callDetailRecordRequest.getWelcomeMessagePromptFlag()) {
+        if (callDetailRecordRequest.getWelcomeMessagePromptFlag() == null) {
             failureReasons.append(String.format(NOT_PRESENT, "welcomeMessagePromptFlag"));
         }
 
         return failureReasons.toString();
     }
 
+    private String validateCallDetailsMobileAcademyElements(CallDetailRecordRequest callDetailRecordRequest) {
+        StringBuilder failureReasons = new StringBuilder();
+
+
+
```

Comment:
Ah, I meant to write a comment there and forgot..but
![escalated](https://cloud.githubusercontent.com/assets/6786794/8788827/203a228a-2ef2-11e5-8244-98db2738329c.gif)